src/dfm/description/match_counter.py

```diff
@@ -0,0 +1,139 @@
+                Defaults to "".
+            label (Optional[str], optional): Label for the spacy match patterns. Helpful when aggregating, e.g. you'd like to count all your male names in a "male names" variable.
+                Defaults to None, indicating each pattern will be labeled by its term.
+            lowercase (Boolean): Whether to match on lowercased tokens or not.
+
+        Returns:
+            List[str]: Spacy match patterns in the shape {"label": [{"LOWER": "term"}]}
+        """
+        out_list = []
+
+        attribute = "LOWER" if lowercase else "TEXT"
+
+        for term in term_list:
+            if label is None:
+                cur_label = label_prefix + term
+            else:
+                cur_label = label_prefix + label
+
+            out_list.append({cur_label: [{attribute: term}]})
+
+        return out_list
+
+    def create_matcher_objects_from_pattern_list(
+        self, pattern_container_list: List[Dict[str, List]]
+    ) -> Matcher:
+        """
+        Generates matcher objects from a list of dictionaries with {matcher_label (str): pattern (list)}
+        Pattern must conform to SpaCy pattern standards:
+
+        Args:
+            pattern_container_list (List[Dict[str, List]]): List of spacy pattern-containers in the shape of
+                {matcher_label (str): pattern (list)}
+
+        Returns:
+            Matcher: Spacy matcher object containing all the patterns from the arg
+
+        Example:
+            >>> pattern_container_list = [
+            >>>    {"atheism": [{"LOWER": {"REGEX": "athei.+"}}]},
+            >>>    {"atheism": [{"LOWER": {"REGEX": "atei.+"}}]},
+            >>>    {"skøde": [{"LOWER": "skøde"}]},
+            >>> ]
+            >>> match_counter.create_matcher_objects_from_pattern_list(pattern_container_list)
+        """
+        matcher_object = Matcher(self.nlp.vocab)
+
+        for pattern_container in pattern_container_list:
+            pattern_label, subpattern_list = list(*pattern_container.items())
+            matcher_object.add(pattern_label, [subpattern_list])
+
+        return matcher_object
+
+    def _get_match_counts_from_doc(self, doc: Doc, matcher_object: Matcher) -> dict:
+        """
+        Get match counts for a list of SpaCy matcher-objects
+
+        args:
+            doc (Doc)
+            matcher_object (Matcher): Object to count from
+
+        returns:
+            A dictionary of the format {pattern_label (str): count (int)}.
+        """
+
+        counts = defaultdict(int)
+
+        # Make sure that all elements are represented in the dict
+        for pattern in matcher_object._patterns:
+            pattern_label = self.nlp.vocab.strings[pattern]
+
+            counts[pattern_label] = 0
+
+        for match_id, start, end in matcher_object(doc):
+            counts[self.nlp.vocab.strings[match_id]] += 1
+
+        return dict(counts)
```

Comment:
Can we more this method down below the `count` method that uses it?

Reply:
Not sure I understand?

Reply:
I think @mollerhoj just wants to move `_get_match_counts_from_doc` below `count` due to their internal relationship 👍

Reply:
Yup, literally talking about how the line position of the methods in the file, nothing complicated really - just what I believe to be standard conventions.